### PR TITLE
feat(doc/code-docs): add checkpoint save/load usage doc

### PR DESCRIPTION
**Pull request by SolenoidWGT**
## Motivation
Add doc for checkpointing.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: doc/code-docs/source/checkpoint.rst [hunk 1]
```diff
@@ -1,11 +1,171 @@
-模型保存
+模型加载与保存
 ===================
 
 InternLM 使用 ``internlm.utils.model_checkpoint.CheckpointManager`` 来管理模型保存。 其中，可以
 使用 ``CheckpointManager.try_save_checkpoint(train_state)`` 来保存指定 step 的模型状态。InternLM支持启动时自动加载最新的模型备份，并在接收信号退出训练时自动进行模型备份。
 
-Checkpointing
+CheckpointManager
```
Comment (huangting4201) [inline doc/code-docs/source/checkpoint.rst:7]: 下划线要大于等于字符的长度，不然会有问题，下面也是

File: doc/code-docs/source/checkpoint.rst [hunk 2]
```diff
@@ -1,11 +1,171 @@
...
+
+- ``async_upload``: 是否开启异步上传。详见 :ref:`asyncupload`
+
+- ``async_upload_tmp_folder``: 异步上传临时存储路径。
+
+- ``oss_snapshot_freq``: 快照存储频率。详见 :ref:`snapshot`
+
+- ``auto_resume``: 是否开启检查点自动恢复。详见 :ref:`autoresume`
+
+- ``stop_file_path`` : 检查点存储控制文件的路径.详见 :ref:`stopfile`
```
Comment (huangting4201) [inline doc/code-docs/source/checkpoint.rst:28]: 路径后应该以中文句号结尾。
另外这些参数，都补充一下默认值吧

File: doc/code-docs/source/checkpoint.rst [hunk 3]
```diff
@@ -1,11 +1,171 @@
...
+      oss_snapshot_freq=int(CHECKPOINT_EVERY / 2),  # snapshot ckpt save frequency.
+  )
+
+
 .. autoclass:: internlm.utils.model_checkpoint.CheckpointManager
     :members:
+
+
+加载与存储格式约定
+-------------
```
Comment (huangting4201) [inline doc/code-docs/source/checkpoint.rst:52]: -------------长度要超出文字内容

File: doc/code-docs/source/checkpoint.rst [hunk 4]
```diff
@@ -1,11 +1,171 @@
...
+- ``async_upload``: 是否开启异步上传。参数类型 ``bool/None``，默认为 ``False``
+
+- ``async_upload_tmp_folder``: 异步上传临时存储路径。参数类型 ``str/None``, 默认值为 ``/dev/shm/{JOB_NAME}_tmp_ckpt/``
+
+需要注意的是，异步上传功能仅在backend为boto3时才会有效果，bcakend为local时只支持同步存储。
+
+``async_upload_tmp_folder`` 设置的的原则为尽量设置为计算节点的local目录，这样才可以获得最佳的异步上传速度，一般来说建议为 ``/dev/shm`` 或 ``/nvme`` 下的路径，如果使用同步上传，则该路径可不给。
+
+
+.. _autoresume:
```
Comment (huangting4201) [inline doc/code-docs/source/checkpoint.rst:121]: 这里注释是不是写反了和下面

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po [hunk 5]
```diff
@@ -36,15 +37,97 @@ msgstr ""
 
 #: ../../source/checkpoint.rst:6
 msgid "InternLM支持启动时自动加载最新的模型备份，并在接收信号退出训练时自动进行模型备份。"
-msgstr "InternLM supports automatic loading of latest ckpt at startup and automatic model checkpointing at signal quit. "
+msgstr ""
```
Comment (huangting4201) [inline doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po:40]: 这段翻译为啥要去掉呢

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po [hunk 6]
```diff
@@ -36,15 +37,97 @@ msgstr ""
...
+"documentation for more details: ref:`stopfile`."
+
+#: ../../source/checkpoint.rst:32
+msgid "下面给出config文件的参数设置例子："
+msgstr "Here is an example of parameter settings in the config file."
 
 #: internlm.utils.model_checkpoint.CheckpointManager:1 of
 msgid "StorageManagerContext"
-msgstr ""
+msgstr "StorageManagerContext"
```
Comment (huangting4201) [inline doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po:130]: 代码块不需要手动翻译，会自动翻译

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po [hunk 7]
```diff
@@ -56,19 +139,19 @@ msgstr ""
 
 #: internlm.utils.model_checkpoint.CheckpointManager.quit_signal_handler of
 msgid "参数"
-msgstr ""
+msgstr "parameter"
 
 #: internlm.utils.model_checkpoint.CheckpointManager.quit_signal_handler of
 msgid "返回"
-msgstr ""
+msgstr "return"
```
Comment (huangting4201) [inline doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po:146]: 这里同上，不需要翻译

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po (hunk 5 above)
Comment (li126com) [reply doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po]: done

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po (hunk 6 above)
Comment (li126com) [reply doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po]: done

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po (hunk 7 above)
Comment (li126com) [reply doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po]: done

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po [hunk 8]
```diff
@@ -20,7 +20,8 @@ msgstr ""
 "Generated-By: Babel 2.12.1\n"
 
 #: ../../source/checkpoint.rst:2
-msgid "模型保存"
+#, fuzzy
```
Comment (zigzagcai) [inline doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po:23]: If there are changes to the original Chinese RST files, the line `fuzzy` translation need to be deleted to make the English document rendered normally.

References: https://github.com/sphinx-doc/sphinx/issues/9618

File: doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po [hunk 9]
```diff
@@ -36,11 +37,93 @@ msgstr ""
...
+#: ../../source/checkpoint.rst:15
+#, fuzzy
+msgid "``save_ckpt_folder``: 检查点存储路径，参数类型 ``str``，默认为： ``None``，在开启检查点存储功能时为必选参数。"
+msgstr ""
+"``save_ckpt_folder``: Checkpoint storage path. Parameter type: ``str``. "
+"This is a required parameter when enabling checkpoint storage "
+"functionality."
+
+#: ../../source/checkpoint.rst:17
+#, fuzzy
```
Comment (zigzagcai) [inline doc/code-docs/locales/en/LC_MESSAGES/checkpoint.po:73]: This line need to be deleted to make english document rendered normally.

References: https://github.com/sphinx-doc/sphinx/issues/9618

**Review by zigzagcai — APPROVED** [review]

**Review by huangting4201 — APPROVED** [review]

**Pull request merged**